compiler-core/src/language_server/tests/definition.rs

```diff
@@ -632,3 +632,129 @@ fn main() {
+    let code = "
+pub type Rec {
+  Var1(Int)
+  Var2(Int, Int)
+}
+
+pub fn make_var() -> Rec {
+  Var1(1)
+}";
+
+    assert_eq!(
+        definition(TestProject::for_source(code), Position::new(6, 22)),
+        Some(Location {
+            uri: Url::from_file_path(Utf8PathBuf::from(if cfg!(target_family = "windows") {
+                r"\\?\C:\src\app.gleam"
+            } else {
+                "/src/app.gleam"
+            }))
+            .unwrap(),
+            range: Range {
+                start: Position {
+                    line: 1,
+                    character: 0
+                },
+                end: Position {
+                    line: 1,
+                    character: 12
+                }
+            }
+        })
+    )
+}
+
+#[test]
+fn goto_definition_type_in_module() {
+    let hex_src = "
+pub type Rec {
+  Var1(Int)
+  Var2(Int, Int)
+}
+";
+
+    let code = "
+import example_module
+fn make_var() -> example_module.Rec {
+  example_module.Var1(1)
+}
+";
+
+    assert_eq!(
+        definition(
+            TestProject::for_source(code).add_hex_module("example_module", hex_src),
+            Position::new(2, 33)
+        ),
+        Some(Location {
+            uri: Url::from_file_path(Utf8PathBuf::from(if cfg!(target_family = "windows") {
+                r"\\?\C:\build\packages\hex\src\example_module.gleam"
+            } else {
+                "/build/packages/hex/src/example_module.gleam"
+            }))
+            .unwrap(),
+            range: Range {
+                start: Position {
+                    line: 1,
+                    character: 0
+                },
+                end: Position {
+                    line: 1,
+                    character: 12
+                }
+            }
+        })
+    )
+}
+
+#[test]
+fn goto_definition_deep_type_in_module() {
+    let hex_src = "
+pub type Wobble {
+  Wobble(Int)
+}
+
+pub type Wibble(a) {
+  Wibble(a)
+}
+
+pub type Wabble(a) {
+  Wabble(a)
+}
+";
+
+    let code = "
+import example_module
+fn make_var() -> example_module.Wabble(example_module.Wibble(example_module.Wobble)) {
+  example_module.Wabble(example_module.Wibble(example_module.Wobble(1)))
+}
+";
+
+    assert_eq!(
+        definition(
+            TestProject::for_source(code).add_hex_module("example_module", hex_src),
+            Position::new(2, 80)
+        ),
+        Some(Location {
+            uri: Url::from_file_path(Utf8PathBuf::from(if cfg!(target_family = "windows") {
+                r"\\?\C:\build\packages\hex\src\example_module.gleam"
+            } else {
+                "/build/packages/hex/src/example_module.gleam"
+            }))
+            .unwrap(),
+            range: Range {
+                start: Position {
+                    line: 1,
+                    character: 0
+                },
+                end: Position {
+                    line: 1,
+                    character: 15
+                }
+            }
+        })
+    )
+}
```

Comment:
Worth having a test for path deps?

Reply:
lemme add one since it'll be easy though i think? it'll use all the same code paths